core/src/test/java/org/apache/calcite/avatica/util/ArrayImplTest.java

```diff
@@ -164,6 +165,267 @@
     }
   }
 
+  /**
+   * The same test as arrayOfStructs(), except we use List instead of ArrayImpl.
+   *
+   * This test deliberately covers the
+   * org.apache.calcite.avatica.util.AbstractCursor.ArrayAccessor.convertValue()
+   * method, which is called from
+   * org.apache.calcite.avatica.util.AbstractCursor.ArrayAccessor.getObject() when
+   * input is a general List instead of org.apache.calcite.avatica.util.ArrayImpl.
+   *
+   * @throws Exception
+   */
+  @Test
+  public void listOfStructs() throws Exception {
+    // Define the struct type we're creating.
```

Comment:
Nitpick: I'd remove the comments like "creating/defining x, y", it is clear from the code itself in most cases